pkg/util/vmopv1/volume.go

```diff
@@ -0,0 +1,547 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package vmopv1
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/vmware/govmomi/object"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha5"
+	cnsv1alpha1 "github.com/vmware-tanzu/vm-operator/external/vsphere-csi-driver/api/v1alpha1"
+)
+
+// These constants define the maximum number of devices that can be attached
+// to each controller type. Unit number 7 is reserved for SCSI controllers
+// as it's used internally by the SCSI controller and cannot be assigned
+// to user devices.
+const (
+	IDEControllerMaxUnitNumber   = 1
```

Comment:
If these constants(including error messages) are not used outside the package, they can just be defined as non export type.